scenes/waka-waka/Scenes/maze.gd

```diff
@@ -1,43 +1,50 @@
 extends TileMapLayer
 
-@export var maze_width: int = 90
-@export var maze_height: int = 90
+@export var maze_size: int = 90
 
-#FIXME: Make rotation and direction names nicer/clearer.
+var maze_data: Array = []
+
+const NORTH: Vector2i = Vector2i(0, -1)
+const EAST: Vector2i = Vector2i(1, 0)
+const SOUTH: Vector2i = Vector2i(0, 1)
+const WEST: Vector2i = Vector2i(-1, 0)
+
+## In NESW order
 const DIRECTIONS: Array = [
-    Vector2i(0, -1), # North
-    Vector2i(1, 0),  # East
-    Vector2i(0, 1),  # South
-    Vector2i(-1, 0)  # West
+    NORTH,
+    EAST,
+    SOUTH,
+    WEST
 ]
 
+## "Normal" rotation for everything is "North", "Up", "Top of screen", Negative-Y axis
+## Think "Rotate to face KEY"
 const TILE_TRANSFORMS: Dictionary = {
-    Vector2i(0, -1): 0, # No rotation, default orientation
-    Vector2i(1, 0): TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_H, # 90 degrees clockwise
-    Vector2i(0, 1): TileSetAtlasSource.TRANSFORM_FLIP_H | TileSetAtlasSource.TRANSFORM_FLIP_V,    # 180 degrees clockwise
-    Vector2i(-1, 0): TileSetAtlasSource.TRANSFORM_FLIP_V | TileSetAtlasSource.TRANSFORM_TRANSPOSE # 270 degrees clockwise
+    NORTH: 0, # No rotation, default orientation
+    EAST: TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_H, # 90 degrees clockwise
+    SOUTH: TileSetAtlasSource.TRANSFORM_FLIP_H | TileSetAtlasSource.TRANSFORM_FLIP_V,   # 180 degrees clockwise
+    WEST: TileSetAtlasSource.TRANSFORM_FLIP_V | TileSetAtlasSource.TRANSFORM_TRANSPOSE  # 270 degrees clockwise
 }
 
-#FIXME: Rename "id" key as "atlas_id".
-
-# Lookup table: maps hashed PackedByteArray (NESW) to (atlas_id, # of clockwise rotations)
+## Lookup table: maps wall locations (NESW) to (atlas_id, # of clockwise rotations) for appropriate piece
```

Comment:
Would love to talk about this on Friday just so I understand what's happening here better